src/mono/mono/eventpipe/ep-rt-mono.h

```diff
@@ -1308,7 +1330,11 @@ inline
 void
 ep_rt_execute_rundown (void)
 {
-	//TODO: Implement.
+	if (ep_rt_config_value_get_rundown () > 0) {
```

Comment:
You should be able to also request rundown on a per-session basis.  This logic will either disable or enable globally only using the env var.

Reply:
This is the same logic as coreclr,. so this is per session at call site, but there is currently an option to disable to globally as well, see https://github.com/dotnet/runtime/blob/69e114c1abf91241a0eeecf1ecceab4711b8aa62/src/coreclr/vm/eventpipesession.cpp#L610 for coreclr C++ implementation and https://github.com/dotnet/runtime/blob/16e6e9530687b34cc93a85d4043093f06f9b7eff/src/coreclr/vm/eventing/eventpipe/ep-rt-coreclr.h#L1877 for coreclr C usage.

Reply:
and here is the logic for this at call site:

https://github.com/dotnet/runtime/blob/e29b83934e7ceef9905f82bd6328cbc72db2278b/src/native/eventpipe/ep.c#L529

so each session can dictate the need for rundown, but the global property can disable it for all (but have no history why that was implemented to begin with).

Reply:
The global flag is most likely a holdover from when EventPipe was a single session started by an environment variable or a config file placed at a known location.  We may want to look at removing that eventually.